qcengine/cli.py

```diff
@@ -4,29 +4,119 @@
+    run_procedure.add_argument('data', type=str, help="A JSON blob of the task description. "
+                                                      "If '-', data will be read from STDIN.")
 
-def main(args=None):
+    args = vars(parser.parse_args())
+    if args["command"] is None:
+        parser.print_help(sys.stderr)
+        exit(1)
+
+    # Print usage and exit if no info options are provided
+    if args["command"] == "info" and not any([v for v in args.values() if v != "info"]):
+        info.print_help()
+        exit(1)
+
+    return args
+
+def info_cli(args):
+
+    def info_version():
+        print(">>> Version information:")
+        print(f"QCEngine version:    {__version__}")
+        try:
+            import qcelemental
+            print(f"QCElemental version: {qcelemental.__version__}")
+        except ImportError as e:
+            print(e)
+        print()
+
+    def info_programs():
+        print(">>> Program information")
+        all_progs = list_all_programs()
+        avail_progs = list_available_programs()
+        print("Available programs:")
+        for prog_name in sorted(avail_progs):
+            version = get_program(prog_name).get_version()
+            if version is None:
+                version = "???"
+            print(f"{prog_name} v{version}")
+
+        print()
+        print("Other supported programs:")
+        print(" ".join(sorted(all_progs - avail_progs)))
+        print()
 
+    def info_procedures():
+        print(">>> Procedure information")
+        all_procs = list_all_procedures()
+        avail_procs = list_available_procedures()
+        print("Available procedures:")
+        for proc_name in sorted(avail_procs):
+            version = get_procedure(proc_name).get_version()
+            if version is None:
+                version = "???"
+            print(f"{proc_name} v{version}")
+
+        print()
+        print("Other supported procedures:")
+        print(" ".join(sorted(all_procs - avail_procs)))
+        print()
+
+    if args["version"] or args["all"]:
+        info_version()
+    if args["programs"] or args["all"]:
+        info_programs()
+    if args["procedures"] or args["all"]:
+        info_procedures()
+    if args["config"] or args["all"]:
+        print(">>> Configuration information")
+        print()
+        print(global_repr())
+
+
+def main(args=None):
     # Grab CLI args if not present
     if args is None:
         args = parse_args()
-
-    ret = compute(json.loads(args["data"]), args["program"])
-    print(json.dumps(ret))
+    command = args.pop("command")
+    if command == "info":
+        info_cli(args)
+    elif command == "run":
+        ret = compute(json.loads(args["data"] if args["data"] != "-" else sys.stdin.read()), args["program"])
+        print(ret.json())
+    elif command == "run-procedure":
+        ret = compute_procedure(json.loads(args["data"] if args["data"] != "-" else sys.stdin.read()), args["procedure"])
+        print(ret.json())
```

Comment:
Should we allow for something like `qcengine run --infile=input.msgpack --outfile=output.msgpack` with support for full extensions? Or is that a bit much for this pass?

Reply:
My target audience for the CLI was demoing/debugging and human interaction. For production input -- MsgPack input can't be generated by hand -- I think the API is the appropriate interface. Let me know if you agree.

Reply:
This could also allow `.yaml` files which would be even easier to write by hand?

`is os.path.isfile(args["data"]); result.parse_file(args["data"]); else ...`

Would allow the user to do what they please.

Reply:
Do we support yaml parsing? 

https://github.com/MolSSI/QCElemental/blob/6ae258fd9ae326b19e9e12cf0dca43d26c6c282d/qcelemental/models/basemodels.py#L56

Reply:
We do not, but using the `parse_file` could be nice as it would be trivial to add there and then it would be picked up everywhere.

Reply:
Added.